api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java

```diff
@@ -0,0 +1,1166 @@
+    CohortDefinition chdVL2 = getPatientsWithVLLessThan1000();
+    CohortDefinition chdVL3 = getPatientsWithMostRecentVLQuantitativeResult();
+    CohortDefinition chdVL4 = getPatientsWhoHaveRegisteredVLLessThan1000ForMoreThan12Months();
+    CohortDefinition chdVL5 = getPatientsWithRecentVLIgualOrGreaterThan1000();
+    CohortDefinition chdVL6 =
+        getPatientsWhoHaveRegisteredVLIgualOrGreaterThan1000ForMoreThan3Months();
+    CohortDefinition chdVL7 = getPatientsWhoDontHaveAnyViralLoad();
+    CohortDefinition chdE1 = txNewCohortQueries.getTxNewBreastfeedingComposition(false);
+    CohortDefinition chdE2 = txNewCohortQueries.getPatientsPregnantEnrolledOnART(false);
+
+    cd.addSearch(
+        "txcurr", EptsReportUtils.map(txcurr, "onOrBefore=${endDate},location=${location}"));
+
+    cd.addSearch("X1", EptsReportUtils.map(chdX1, "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "X2",
+        EptsReportUtils.map(
+            chdX2, "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "X3",
+        EptsReportUtils.map(
+            chdX3, "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "VL1",
+        EptsReportUtils.map(
+            chdVL1, "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch("VL2", EptsReportUtils.map(chdVL2, "startDate=${startDate},location=${location}"));
+    cd.addSearch("VL3", EptsReportUtils.map(chdVL3, "startDate=${startDate},location=${location}"));
+    cd.addSearch(
+        "VL4",
+        EptsReportUtils.map(
+            chdVL4, "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch("VL5", EptsReportUtils.map(chdVL5, "startDate=${startDate},location=${location}"));
+    cd.addSearch(
+        "VL6",
+        EptsReportUtils.map(
+            chdVL6, "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch("VL7", EptsReportUtils.map(chdVL7, "startDate=${startDate},location=${location}"));
+    cd.addSearch(
+        "E1",
+        EptsReportUtils.map(
+            chdE1, "onOrAfter=${startDate},onOrBefore=${endDate},location=${location}"));
+    cd.addSearch(
+        "E2",
+        EptsReportUtils.map(
+            chdE2, "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString(
+        "txcurr AND (X1 OR X2 OR X3) AND VL1 (( (VL2 OR VL3) AND VL4) OR (VL5 AND VL6) OR VL7) AND NOT (E1 OR E2)");
+
+    return cd;
+  }
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   *
+   * <p>X1- select all patients with “Data da próxima consulta” (concept id 1410, value_datetime) as
+   * “Last Next scheduled Consultation Date” from the most recent FICHA CLÍNICA (encounter type 6,9)
+   * by report end date (encounter_datetime <= endDate) and “Last Next scheduled Consultation Date”
+   * >= startDate and <= endDate
```

Comment:
Ni @eliazmuchave, I dont see this section in the query

Reply:
@psmatsinhe, I fixed it